src/blockchain/utils/blkfile.rs

```diff
@@ -42,21 +42,29 @@ impl BlkFile {
         let blk_ext = String::from(".dat");
 
         for entry in content {
-            if let Ok(e) = entry {
-                // Check if it's a file
-                if try!(e.file_type()).is_file() {
-                    let file_name = String::from(transform!(e.file_name().to_str()));
-                    // Check if it's a valid blk file
-                    if let Some(index) = BlkFile::parse_blk_index(&file_name, &blk_prefix, &blk_ext) {
-                        // Only process new blk files
-                        if index >= min_blk_idx {
-                            // Build BlkFile structures
-                            let file_len = try!(e.metadata()).len();
-                            trace!(target: "blkfile", "Adding {}... (index: {}, size: {})", e.path().display(), index, file_len);
-                            blk_files.push(BlkFile::new(e.path(), index, file_len));
-                        }
+            if let Ok(de) = entry {
+
+                let sl = de.file_type().unwrap().is_symlink();
+                let mut path : PathBuf = de.path();
+                let mut metadata :Metadata = de.metadata().unwrap();
+                if sl {
+                    path = fs::read_link(path.clone()).unwrap();
+                    metadata = fs::metadata(path.clone()).unwrap();
+                }
+
+                let file_name = String::from(transform!(path.as_path().file_name().unwrap().to_str()));
+
+                // Check if it's a valid blk file
+                if let Some(index) = BlkFile::parse_blk_index(&file_name, &blk_prefix, &blk_ext) {
```

Comment:
Is the `is_file()` check no longer necessary?, what happens when `file_name` is a directory?

Reply:
reintroducing the `is_file()` check in another PR